### ISSUE TITLE
Update won't update

### ISSUE THREAD

**Issue by rappen** [MscrmTools/XrmToolBox#1356]
Trying to update to latest version, looks good, but isn't updated...
See https://app.screencast.com/ncZTdCRsTidXW

Unsure if this is related, but I'm running it as a custom overridepath.

Downloading and unpacking works fine.

---
- Connection Controls Version: 1.2023.6.56
- XrmToolBox Version: 1.2023.12.68

**Comment by MscrmTools** [comment]
Well, I'm overriding path too and I always test to update as soon as I release a new version (as I'm not using dev version as my daily to use version) and it worked just fine for me...

Anyway, I will double check this tomorrow.

**Comment by rappen** [comment]
On my other laptop, it updated correctly... No idea what is different :-/
"the other one" is the new one; on that, the WebView2 doesn't work. Weird it is...

**Comment by MscrmTools** [comment]
WebView2 does not work either on my computer...

But I do have an assembly binding redirection for this assembly and it never had issues before...

**Comment by rappen** [comment]
What?? Now WebView2 works 😂🤔
https://app.screencast.com/zjJpuuIWms1Sl

**Comment by pietermcornelius** [comment]
i updated this morning as none of my connections wanted to work, now I can connect but SQL for CDS is not working. I installed it and when I try and install it downloads but does not install

**Comment by viacheslav-protsenko** [comment]
> i updated this morning as none of my connections wanted to work, now I can connect but SQL for CDS is not working. I installed it and when I try and install it downloads but does not install

the same

**Comment by MarkMpn** [comment]
> i updated this morning as none of my connections wanted to work, now I can connect but SQL for CDS is not working. I installed it and when I try and install it downloads but does not install

I can reproduce an error installing SQL 4 CDS if I uninstall it and they try to install it again - the file copying process fails due to the files in the Plugins\MarkMpn.Sql4Cds subfolder being in use. Installing from a fresh install succeeds. I can avoid the problem by deleting the Plugins\MarkMpn.Sql4Cds folder entirely before attempting to install it.

This seems like a gap in the uninstall process - ideally that subfolder should be deleted when uninstalling the tool as well as the main dll.

**Comment by SPWareing** [comment]
> i updated this morning as none of my connections wanted to work, now I can connect but SQL for CDS is not working. I installed it and when I try and install it downloads but does not install



I updated this morning and all my connections stopped working.

Ended up replacing the updated DLLs with a previous release

**Comment by pietermcornelius** [comment]
> > i updated this morning as none of my connections wanted to work, now I can connect but SQL for CDS is not working. I installed it and when I try and install it downloads but does not install
> 
> I can reproduce an error installing SQL 4 CDS if I uninstall it and they try to install it again - the file copying process fails due to the files in the Plugins\MarkMpn.Sql4Cds subfolder being in use. Installing from a fresh install succeeds. I can avoid the problem by deleting the Plugins\MarkMpn.Sql4Cds folder entirely before attempting to install it.
> 
> This seems like a gap in the uninstall process - ideally that subfolder should be deleted when uninstalling the tool as well as the main dll.

Thank you @MarkMpn  will try this now I got the SSMS version working also

**Comment by pietermcornelius** [comment]
> > > i updated this morning as none of my connections wanted to work, now I can connect but SQL for CDS is not working. I installed it and when I try and install it downloads but does not install
> > 
> > 
> > I can reproduce an error installing SQL 4 CDS if I uninstall it and they try to install it again - the file copying process fails due to the files in the Plugins\MarkMpn.Sql4Cds subfolder being in use. Installing from a fresh install succeeds. I can avoid the problem by deleting the Plugins\MarkMpn.Sql4Cds folder entirely before attempting to install it.
> > This seems like a gap in the uninstall process - ideally that subfolder should be deleted when uninstalling the tool as well as the main dll.
> 
> Thank you @MarkMpn will try this now I got the SSMS version working also

Sorted followed your steps

**Comment by MscrmTools** [comment]
> > i updated this morning as none of my connections wanted to work, now I can connect but SQL for CDS is not working. I installed it and when I try and install it downloads but does not install
> 
> I updated this morning and all my connections stopped working.
> 
> Ended up replacing the updated DLLs with a previous release

When you say "all my connections stopped working". What are the symptoms? Any error message?

**Comment by SPWareing** [comment]
> > > i updated this morning as none of my connections wanted to work, now I can connect but SQL for CDS is not working. I installed it and when I try and install it downloads but does not install
> > 
> > I updated this morning and all my connections stopped working.
> > 
> > Ended up replacing the updated DLLs with a previous release
> 
> When you say "all my connections stopped working". What are the symptoms? Any error message?

I was unable to use any of my stored connections or create new ones.

Error messages varied from

- Cache error
- Dataverse Error 

Among others.

I just looked at the updated DLLs from the connection update and replaced them with the previous versions.

I'm not keen on repeating the error by updating the connection package again

**Comment by SPWareing** [comment]
> But you do use latest version of XrmToolBox, right? Can you show me the version of assemblies (Settings > Assemblies)

No I wasn't, I was using v1.2023.12.68 

I've just upgraded to the new release and now connection issues

I was prompted to update the connections in the previous release which caused the issues.

**Comment by MscrmTools** [comment]
Yes I know... I should avoid prompting for new connection controls if a new version of XrmToolBox is available
